### PR TITLE
MapboxOnboardRouter: fix RouteUrl creating, incorrect waypoints

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

MapboxOnboardRouter: Fix RouteUrl creating, incorrect `waypoints`

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Could you add a test @RingerJK?

**Comment by RingerJK** [comment]
> Could you add a test @RingerJK?

sure!

File: libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt [hunk 1]
```diff
@@ -89,21 +89,20 @@ class MapboxOnboardRouter : Router {
...
+                waypoints = waypoints,
                 destination = destination,
-                steps = routeOptions.steps() ?: false,
-                voiceIntruction = routeOptions.voiceInstructions() ?: false,
-                bannerIntruction = routeOptions.bannerInstructions() ?: false,
-                roundaboutExits = routeOptions.roundaboutExits() ?: false
+                steps = routeOptions.steps() ?: RouteUrl.STEPS_DEFAULT_VALUE,
+                voiceInstruction = routeOptions.voiceInstructions() ?: RouteUrl.VOICE_INSTRUCTION_DEFAULT_VALUE,
+                bannerInstruction = routeOptions.bannerInstructions() ?: RouteUrl.BANNER_INSTRUCTION_DEFAULT_VALUE,
+                roundaboutExits = routeOptions.roundaboutExits() ?: RouteUrl.ROUND_ABOUTEXITS_DEFAULT_VALUE
```
Comment (korshaknn) [inline libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt]:
```suggestion
                roundaboutExits = routeOptions.roundaboutExits() ?: RouteUrl.ROUNDABOUT_EXITS_DEFAULT_VALUE
```

Comment (LukasPaczos) [inline libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt]: Looking at the directions api documentation https://docs.mapbox.com/api/navigation/, the default value is `false` and we should probably stick to it.

Comment (RingerJK) [reply libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt]: thanks! fixed

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt [hunk 2]
```diff
@@ -34,6 +34,12 @@ class RouteUrl(
         private const val QUERY_PARAM_ROUNDABOUT_EXITS = "roundabout_exits"
         private const val QUERY_PARAM_ENABLE_REFRESH = "enable_refresh"
 
+        const val STEPS_DEFAULT_VALUE = false
+        const val VOICE_INSTRUCTION_DEFAULT_VALUE = false
+        const val BANNER_INSTRUCTION_DEFAULT_VALUE = false
+        const val ROUNDABOUT_EXITS_DEFAULT_VALUE = false
+        const val ENABLE_REFRESH_DEFAULT_VALUE = true
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: Should this be true?

/cc @Guardiola31337

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: Currently we're enabling refresh by default 👀 https://github.com/mapbox/mapbox-navigation-android/blob/c3d1f4cebb2d57a71bfbb6c4bf8953e9da7eca98/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt#L742 but we can definitely discuss if it makes sense now with the new setup.

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: The same thing happens with steps, voice and banner instructions and roundabout exits 👀 https://github.com/mapbox/mapbox-navigation-android/blob/c3d1f4cebb2d57a71bfbb6c4bf8953e9da7eca98/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt#L735

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: Also want to note that the Navigation use case is different from the Directions one i.e. the defaults could be different (that's why we had some of them unmatched and always default to what makes most sense in Navigation).

Also, it's difficult to know what's the default value from the name e.g. `ENABLE_REFRESH_DEFAULT_VALUE` and it's hard to tell when reading the code. For understandability, should we rename it into something more explicit?

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: @Guardiola31337 you mean like `ENABLE_REFRESH_DEFAULT_VALUE ` -> `ENABLE_REFRESH_DEFAULT_VALUE_FALSE`

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: I'd even remove the `DEFAULT` so something like this `REFRESH_ENABLED` / `REFRESH_DISABLED` (in this particular case I'd also remove the `ENABLE` at the beginning). What do you think? I think it reads 👌

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: I think it's intension to hide value under the name of value 😃 
suggest next naming: `REFRESH_INITIAL = false/true`

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: We'd be in the same situation, we wouldn't know when reading the code. BTW what's the purpose of hiding the value? It's an internal default value so the more context the better, for readability and understandability.

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: if we'd like to change this value we should change name of it as well, it's not the best approach. btw it's already public, not hidden, value. 
To see value you need to clic on it, open docs or class where this value is, but not ready VALUE from name, it's strange at least, intread we could just write value directly, without saving to property/field

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: All of these constants should be `internal`, don't see any value of exposing them publicly, as they're only used internally for handling the default values. You can look at the current implementation and they're actually `internal` e.g. https://github.com/mapbox/mapbox-navigation-android/blob/f6963eeff0b84c0846026bb833074badc06fa988/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationConstants.kt#L140

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: they could be internal, I'm agree

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: these fields are used in different module, so cannot be internal (are used in `MapboxOnboardRouter`)

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: @RingerJK :eyes: https://github.com/mapbox/mapbox-navigation-android/issues/2417

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: @Guardiola31337 got it, thanks. Need sync. I think it works good for whole class, but if we should hide `field` only, it's not working so pure

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: Not sure if I follow / what you mean by `it's not working so pure` but yeah let's sync up internally :+1:

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: btw for `RouteUrl` it works

Comment (Guardiola31337) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: What does it work for `RouteUrl`? :thinking:

Comment (RingerJK) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/route/RouteUrl.kt]: @Guardiola31337 [PR](https://github.com/mapbox/mapbox-navigation-android/pull/2426)

**Pull request merged**